vue.config.js

```diff
@@ -45,6 +45,19 @@ module.exports = {
 					},
 				})])
 		}
+
+		// watch needs to be in poll mode for it to work properly in Vagrant
+		// set watch options for dev server mode (e.g. npm run serve)
+		config.devServer.watchOptions({
+			poll: 1500,
+			ignored: [/node_modules/],
+		})
+
+		// set watch options for build mode with watch enabled (e.g. npm run watch)
```

Comment:
Could you explain why we have distinct build mode? Also how do we update/restart the server if node_modules is changed?

Reply:
I added the "watch" and "dev-watch" commands when we thought that we would just build the app in watch mode and let nginx serve the files, but it turns out that you miss some useful development features that way. So they are probably not needed if we keep using the dev server mode. (Removing them would also make the addition of config.watchOptions unnecessary.)

I currently run the front-end in the development environment manually on the Vagrant machine, so I can simply restart it (e.g. Ctrl+C, npm run serve). I also tried not ignoring node_modules but then the polling had around 50% CPU usage on my computer.

Reply:
Okey, sounds good. Good to know that it is that easy to restart the server. For some reason I thought that it would be running in docker. I also agree that it is better to keep node_modules out of watch scope.

Reply:
Yes, you'd probably miss out on the direct socket for hot reload.

Is it possible to detect whether you're running on a fully installed virtual server environment or on your own local machine, so we can only enable the polling for virtual machines?